python/ipywidgets/ipywidgets/widgets/widget.py

```diff
@@ -686,7 +693,7 @@ def notify_change(self, change):
         # Send the state to the frontend before the user-registered callbacks
         # are called.
         name = change['name']
-        if self.comm is not None and self.comm.kernel is not None:
+        if self.comm is not None and (self.comm.kernel is not None if hasattr(self.comm, "kernel") else True):
```

Comment:
```suggestion
        if self.comm is not None and getattr(self.comm, 'kernel', True) is not None:
```
do you think this look better.. ?